src/batchpage.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import Dropdown from './dropdown.js'
+import Navbar from './navbar.js'
+import CollectionsBox from './collections-box.js'
+import Batch from './batch.js'
+import parseQuery from './parseQuery.js'
+
+class BatchPage extends React.Component {
+
+  constructor(props) {
+    super(props)
+    var args = parseQuery(window.location.href)
+    this.state = {
+      terms: args.terms ? ("" + args.terms).replace(/\+/g, ' ') : "",
+      args: args,
+      collections: null,
+      selectedCollections: null,
+      limits: null,
+    }
+  }
+
+  componentDidMount() {
+    this.getCollectionsAndLimits()
+  }
+
+  getCollectionsAndLimits() {
+    $.get('/collections', function (resp) {
+      var collections = resp
+      var selectedCollections = null
+      var args = this.state.args
+      if (args.collections) {
+        var sel = '' + args.collections
+        selectedCollections = sel ? sel.split('+') : []
+      }
+      else {
+        selectedCollections = resp.map((c) => c.name)
+      }
+
+      this.setState({
+        collections,
+        selectedCollections,
+      })
+    }.bind(this))
+
+    $.get('/limits', function(resp) {
```

Comment:
maybe poll this every `limits.requests.limit / 3` or something, so the user won't get blocked out if you disable the button below